specs/logging_implementation.adoc

```diff
@@ -0,0 +1,174 @@
+= Specification for logging implementation
+
+Logs and alerts need to be treated as messages emitted by a component.
+
+== Problem description
+
+* Logs and alerts need to be treated as messages emitted by a component.
+* A schema needs to be defined for emitted messages. This could be a json schema.
+  The schema needs to include metadata that allows the exact point of origin of
+  the message to be known.
+* Message priorities, types etc. need to be defined at the time of emission.
+* The integration/bridge applications must parse and restructure messages emitted
+  by the storage system to fit in the tendrl message schema before emitting it.
+* For messages that serve as notifications to the administrator (or alerts),
+  exact error message must be hardcoded by the developers. No backtraces or
+  system generated error messages should be sent as notifications to the
+  administrator.
+* Notifications sent to the administrator should provide sufficient context to
+  enable debugging of the problem via investigation into the log files.
+* Notifications and alerts need to be delivered to the node agent alert sockets.
+* Commons needs to implement a logging library that hides all the implementation
+  details and configuration options from any of the components using the library.
+  Components should be able to blindly emit messages to the common logging module
+  and it should be able to deliver them to either syslog, log files, alert
+  sockets etc. depending upon the metadata and configuration.
+* Syslog must be the default logging configuration. Logging to individual
+  configuration files should be thoroughly discouraged.
+
+== Use Cases
+
+Logs and alerts need to be treated as messages.
+
+== Proposed change
+* Create a new class called "Message" in common with attributes name, type,
```

Comment:
I feel we would need info about the module which is raising this message so that module specific logging configuration can be used for logging the messages. I am coming from a point where each module like node_agent, ceph_integration, gluster_integartion etc can have their own log-levels and loggers defined.
Does not look very clear how this scenario will be taken care with this model..